modules/rgbd/samples/large_kinfu_demo.cpp

```diff
@@ -0,0 +1,528 @@
+                float cubeSize = 1.f;
+                params.volumeParams.voxelSize = cubeSize/params.volumeParams.resolution[0];
+                params.volumeParams.tsdfTruncDist = 0.01f;
+                params.icpDistThresh = 0.01f;
+                params.volumeParams.pose = Affine3f().translate(Vec3f(-cubeSize/2.f,
+                                                               -cubeSize/2.f,
+                                                               0.05f));
+                params.truncateThreshold = 2.5f;
+                params.bilateral_sigma_depth = 0.01f;
+            }
+
+            if(sourceType == Type::DEPTH_KINECT2)
+            {
+                Matx<float, 1, 5> distCoeffs;
+                distCoeffs(0) = Kinect2Params::k1;
+                distCoeffs(1) = Kinect2Params::k2;
+                distCoeffs(4) = Kinect2Params::k3;
+
+                initUndistortRectifyMap(camMatrix, distCoeffs, cv::noArray(),
+                                        camMatrix, frameSize, CV_16SC2,
+                                        undistortMap1, undistortMap2);
+            }
+        }
+    }
+
+    vector<string> depthFileList;
+    size_t frameIdx;
+    VideoCapture vc;
+    UMat undistortMap1, undistortMap2;
+    Type sourceType;
+};
+
+#ifdef HAVE_OPENCV_VIZ
+const std::string vizWindowName = "cloud";
+
+struct PauseCallbackArgs
+{
+    PauseCallbackArgs(LargeKinfu& _largeKinfu) : largeKinfu(_largeKinfu)
+    { }
+
+    LargeKinfu& largeKinfu;
+};
+
+void pauseCallback(const viz::MouseEvent& me, void* args);
+void pauseCallback(const viz::MouseEvent& me, void* args)
+{
+    if(me.type == viz::MouseEvent::Type::MouseMove       ||
+       me.type == viz::MouseEvent::Type::MouseScrollDown ||
+       me.type == viz::MouseEvent::Type::MouseScrollUp)
+    {
+        PauseCallbackArgs pca = *((PauseCallbackArgs*)(args));
+        viz::Viz3d window(vizWindowName);
+        UMat rendered;
+        pca.largeKinfu.render(rendered, window.getViewerPose().matrix);
+        imshow("render", rendered);
+        waitKey(1);
+    }
+}
+#endif
+
+static const char* keys =
+{
+    "{help h usage ? | | print this message   }"
+    "{depth  | | Path to depth.txt file listing a set of depth images }"
+    "{camera |0| Index of depth camera to be used as a depth source }"
+    "{coarse | | Run on coarse settings (fast but ugly) or on default (slow but looks better),"
+        " in coarse mode points and normals are displayed }"
+    "{useHashTSDF | | Use the newer hashtable based TSDFVolume (relatively fast) and for larger reconstructions}"
+    "{idle   | | Do not run KinFu, just display depth frames }"
+    "{record | | Write depth frames to specified file list"
+        " (the same format as for the 'depth' key) }"
+};
+
+static const std::string message =
+ "\nThis demo uses live depth input or RGB-D dataset taken from"
+ "\nhttps://vision.in.tum.de/data/datasets/rgbd-dataset"
+ "\nto demonstrate KinectFusion implementation \n";
+
+
```

Comment:
Do you still think, we should include both in the same demo application?

Reply:
We can leave `large_kinfu_demo` if it's going to be changed radically: different threading model or different vizualization, or something else.
In its current state it makes more sense to merge it to `kinfu_demo` since there's no significant differences between them.

Reply:
Let it be a separate demo for future rework